src/main/java/uk/gov/hmcts/reform/sscs/robotics/json/RoboticsJsonMapper.java

```diff
@@ -25,22 +27,29 @@
     private static final String ESA_CASE_CODE = "051DD";
     private static final String PIP_CASE_CODE = "002DD";
 
-    private final boolean readyToListFeatureEnabled;
-
-    private DwpAddressLookupService dwpAddressLookupService;
+    private final DwpAddressLookupService dwpAddressLookupService;
+    private final RegionalProcessingCenterService regionalProcessingCenterService;
+    private final AirLookupService airLookupService;
 
     @Autowired
-    public RoboticsJsonMapper(@Value("${robotics.readyToList.feature}") boolean readyToListFeatureEnabled,
-                              DwpAddressLookupService dwpAddressLookupService) {
-        this.readyToListFeatureEnabled = readyToListFeatureEnabled;
+    public RoboticsJsonMapper(DwpAddressLookupService dwpAddressLookupService,
+                              RegionalProcessingCenterService regionalProcessingCenterService,
+                              AirLookupService airLookupService) {
         this.dwpAddressLookupService = dwpAddressLookupService;
+        this.regionalProcessingCenterService = regionalProcessingCenterService;
+        this.airLookupService = airLookupService;
     }
 
     public JSONObject map(RoboticsWrapper roboticsWrapper) {
 
         SscsCaseData sscsCaseData = roboticsWrapper.getSscsCaseData();
 
-        JSONObject obj = buildAppealDetails(new JSONObject(), sscsCaseData, roboticsWrapper.getVenueName());
+        String firstHalfOfPostcode = regionalProcessingCenterService.getFirstHalfOfPostcode(sscsCaseData.getAppeal().getAppellant().getAddress().getPostcode());
```

Comment:
We are already using same logic in tribunal-api (SubmitAppealService) and creating sscsCaseData before sending to robotics